tests/test_build_qthelp_leanid.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+"""
+    test_build_qthelp_leanid
+    ~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Test the Qt Help builder and check its output.  We don't need to
+    test the HTML itself; that's already handled by
+    :file:`test_build_html.py`. Test for id generation in qhp files
+
+    :copyright: Copyright 2007-2017 by the Sphinx team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+import pytest
+
+
+@pytest.mark.sphinx('qthelp', testroot='index')
+def test_qthelp_namespace(app, status, warning):
+    # default namespace
+    app.builder.build_all()
+    qhp = (app.outdir / 'Python.qhp').text()
+    assert '<keyword name="features (ID_FEATURE)" id="ID_FEATURE.features"' in qhp
```

Comment:
Do you know what purpose this id parameter is used? (I've never used qthelp...)
I'd like to know why have we assigned IDs manually for the keyword tag.

And I also want to know why `qthelp_id_lean` option is needed. Is there any reason to choose IDs lean or not?

Reply:
My purpose is to open Context Specific Online Help from my application from the code. This supports en, de, fr Online Help depending on the language settings
There are too possibilities: Too look for keywords, they are normaly generated from Document filenames, and keywords from index. But they are are language specific.

So i have too options: 
in my code a can call linksForKeyword () and look for keywords with prior tr translation from qt. The main disadvantage is that you have too keep this in mind if you change the indexes.
Second is to use the ID for index entries.
This id should be a unique and language independent identifier.
So you can manually generate index entries in 3 different language rst documents and then use
linksForIdentifier() in the code

In the current implementation of sphinx everything gets now mixed. So the id contains the keyword and i have now language dependencies again. So the idea is to separate this clearly.

Cause my implementation breaks current behaviour this patch would make current implementations to fail. So i think its better to use an option for this and make the old behaviour deprecated to give people a chance to change implementations.